src/Core/Exporters/Concrete/BaseExporter.cs

```diff
@@ -0,0 +1,122 @@
+﻿// <copyright file="BaseExporter.cs" company="Microsoft">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// </copyright>
+
+namespace Core.Exporters.Concrete
+{
+    using System;
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading.Tasks;
+    using Core.Configurations.Exporters;
+    using Core.Exporters.Abstract;
+    using Core.Extensions;
+    using Core.Providers;
+    using Core.Utils;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using Prometheus;
+
+    public abstract class BaseExporter : IExporter
+    {
+        protected readonly IContentProvider ContentProvider;
+        protected readonly IPrometheusUtils PrometheusUtils;
+        protected readonly BaseExporterConfiguration BaseConfiguration;
+        protected readonly ILogger Logger;
+        protected readonly Type ComponentType;
+
+        private readonly Dictionary<string, string> _exporterMetricsLabels;
+
+        protected BaseExporter(
+            IContentProvider contentProvider,
+            IPrometheusUtils prometheusUtils,
+            BaseExporterConfiguration baseConfiguration,
+            Type componentType,
+            ILogger logger)
+        {
+            ContentProvider = contentProvider;
+            PrometheusUtils = prometheusUtils;
+            BaseConfiguration = baseConfiguration;
+            ComponentType = componentType;
+            Logger = logger;
+            Collectors = new ConcurrentDictionary<string, Collector>();
+
+            _exporterMetricsLabels = new Dictionary<string, string>()
+            {
+                { "Exporter", $"{GetType().Name}" }
+            };
+            _exporterMetricsLabels.TryAdd(BaseConfiguration.DefaultLabels);
+        }
+
+        /// <inheritdoc />
+        public ConcurrentDictionary<string, Collector> Collectors { get; protected set; }
+
+        /// <inheritdoc />
+        public async Task ExportMetricsAsync(string endpointUrlSuffix = null)
+        {
+            // Validating suffix
+            var fullEndpointUri = BaseConfiguration.UriEndpoint;
```

Comment:
If you'll move the setting of the fullEndpointUri  to its own helper method you'd be able to get much tighter code (single return statement)

Reply:
Refactored accordingly.